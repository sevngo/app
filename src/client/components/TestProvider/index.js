import { MuiThemeProvider } from '@material-ui/core';
import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter, Route } from 'react-router-dom';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import reducer from '../../reducers';
import { theme } from '../../utils/theme';

jest.mock('../../store');

const TestProvider = ({
  children,
  initialState = {},
  router: { initialEntries, path } = {},
}) => {
  const store = createStore(reducer, initialState, applyMiddleware(thunk));
  return (
    <Provider store={store}>
      <IntlProvider locale="en" messages={{}}>
        <MemoryRouter initialEntries={initialEntries}>
          <Route path={path}>
            <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
          </Route>
        </MemoryRouter>
      </IntlProvider>
    </Provider>
  );
};

export default TestProvider;
