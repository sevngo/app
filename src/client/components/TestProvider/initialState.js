const initialState = {
  auth: {
    _id: '6008432b6ed92c2ac837ec6c',
    username: 'asdasd',
    email: 'asdasd@asdasd.com',
    gender: 'female',
    birthDate: '1991-01-20T00:00:00.000Z',
    address: {
      name: '5 All√©e Claude Chastillon, 93290 Tremblay-en-France, France',
      type: 'Point',
      coordinates: [2.5654428, 48.955157299999996],
    },
    notifications: [
      {
        user: {
          _id: '5caf003feaca23291c0cec34',
          username: 'qweqwe',
        },
        createdAt: '2020-01-23T14:10:55.857Z',
        _id: '5e29a96f7e701b58a77dcdbb',
        messageId: 'friendLogged',
      },
      {
        user: {
          _id: '5caf003feaca23291c0cec36',
          username: 'asdasd',
        },
        createdAt: '2020-01-23T14:11:55.857Z',
        _id: '5e29a96f7e701b58a77dcdba',
        messageId: 'friendLogged',
      },
    ],
    token:
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI2MDA4NDMyYjZlZDkyYzJhYzgzN2VjNmMiLCJpYXQiOjE2MTEyNTc1MjQsImV4cCI6MTYxMTM0MzkyNH0.suG2IyrXwsI3bOcK1jG7XpasHfwQM1Fip3YrUzX01Pc',
  },
  users: {
    filter: {
      gender: 'male',
      maxDistance: 20000,
      ageRange: [18, 50],
      sortBy: 'distance:asc',
      limit: 10,
      skip: 0,
    },
    data: [
      {
        _id: '60084a46203c4e342b13114c',
        username: 'Shanie21',
        birthDate: '1998-03-03T13:43:52.868Z',
        gender: 'male',
        address: {
          name: '247 Bert Village',
        },
        distance: 2640.691228895585,
      },
      {
        _id: '60084a46203c4e342b13113c',
        username: 'Brianne82',
        birthDate: '1990-05-11T19:00:30.586Z',
        gender: 'male',
        address: {
          name: '4860 Berenice Lodge',
        },
        distance: 2993.4424127582033,
      },
      {
        _id: '60084a46203c4e342b131140',
        username: 'Jon3',
        birthDate: '1973-05-08T21:14:45.718Z',
        gender: 'male',
        address: {
          name: '259 Fabiola Station',
        },
        distance: 6966.4502984182745,
      },
      {
        _id: '60084a46203c4e342b131139',
        username: 'Joannie_Corkery92',
        birthDate: '1998-09-22T09:23:53.485Z',
        gender: 'male',
        address: {
          name: '2431 Marisol Park',
        },
        distance: 9156.496213903085,
      },
      {
        _id: '60084a46203c4e342b13113a',
        username: 'Lilly.MacGyver',
        birthDate: '2002-05-18T07:47:56.337Z',
        gender: 'male',
        address: {
          name: '27384 Carroll Rue',
        },
        distance: 10229.485636305038,
      },
      {
        _id: '60084a46203c4e342b131143',
        username: 'Lacy_Harber23',
        birthDate: '2001-11-18T21:39:20.224Z',
        gender: 'male',
        address: {
          name: '1810 Eliseo Key',
        },
        distance: 11798.426998259116,
      },
      {
        _id: '60084a46203c4e342b131147',
        username: 'Sophie.Gorczany',
        birthDate: '1992-08-23T18:55:03.359Z',
        gender: 'male',
        address: {
          name: '149 Koch Vista',
        },
        distance: 13149.980680446492,
      },
      {
        _id: '60084a46203c4e342b131142',
        username: 'Joelle.Weissnat',
        birthDate: '1985-02-24T06:13:19.138Z',
        gender: 'male',
        address: {
          name: '74795 Howe Inlet',
        },
        distance: 16796.075505426317,
      },
    ],
    total: 8,
  },
  user: {
    _id: '60084a46203c4e342b13114c',
    username: 'Shanie21',
    birthDate: '1998-03-03T13:43:52.868Z',
    gender: 'male',
    address: {
      name: '247 Bert Village',
    },
  },
  loading: true,
  snackbar: {
    variant: 'success',
  },
};

export default initialState;
