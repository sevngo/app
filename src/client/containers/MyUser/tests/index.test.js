import {
  fireEvent,
  render,
  waitFor,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import axios from '../../../api';
import React from 'react';
import TestProvider from '../../../components/TestProvider';
import Component from '../index';

global.google = {
  maps: {
    places: {
      Autocomplete: function () {
        return {
          setTypes: jest.fn(),
          setFields: jest.fn(),
          addListener: jest.fn(),
        };
      },
    },
  },
};

jest.mock('../../../api');

describe('MyUser', () => {
  const authId = '6008432b6ed92c2ac837ec6c';
  const initialState = {
    auth: {
      _id: '6008432b6ed92c2ac837ec6c',
      username: 'asdasd',
      email: 'asdasd@asdasd.com',
      gender: 'female',
      birthDate: '1991-01-20T00:00:00.000Z',
      address: {
        name: '5 Allée Claude Chastillon, 93290 Tremblay-en-France, France',
        type: 'Point',
        coordinates: [2.5654428, 48.955157299999996],
      },
      token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9',
      usersLiked: [{ _id: '6008432b6ed92c2ac837ec6c' }],
    },
  };

  it('should be able to click on open file', async () => {
    const { getByTestId } = render(
      <TestProvider initialState={initialState}>
        <Component />
      </TestProvider>
    );
    fireEvent.click(getByTestId('openFile'));
  });

  it('should uploadFile', async () => {
    const { getByTestId, findByTestId } = render(
      <TestProvider initialState={initialState}>
        <Component />
      </TestProvider>
    );
    getByTestId('imageUnavailable');
    axios.post.mockResolvedValue({ data: { image: 'a' } });
    const file = new File(['(⌐□_□)'], 'chucknorris.png', { type: 'image/png' });
    fireEvent.change(getByTestId('uploadFile'), { target: { files: [file] } });
    await waitFor(() =>
      expect(axios.post).toHaveBeenCalledWith(
        `/api/users/${authId}/image`,
        expect.any(FormData)
      )
    );
    await findByTestId('imageAvailable');
  });

  it('should update my user', async () => {
    const { getByRole } = render(
      <TestProvider initialState={initialState}>
        <Component />
      </TestProvider>
    );
    fireEvent.click(getByRole('radio', { name: 'Male' }));
    axios.patch.mockResolvedValue({ data: { gender: 'male' } });
    fireEvent.click(getByRole('button', { name: 'Submit' }));

    await waitFor(() =>
      expect(axios.patch).toHaveBeenCalledWith(
        `/api/users`,
        {
          address: {
            coordinates: [2.5654428, 48.955157299999996],
            name: '5 Allée Claude Chastillon, 93290 Tremblay-en-France, France',
            type: 'Point',
          },
          birthDate: '1991-01-20',
          email: 'asdasd@asdasd.com',
          gender: 'male',
          username: 'asdasd',
        },
        { headers: { Authorization: 'Bearer undefined' } }
      )
    );
    await waitFor(() => getByRole('button', { name: 'Submit' }).disabled);
  });

  it('should dislike user', async () => {
    const { getByRole } = render(
      <TestProvider initialState={initialState}>
        <Component />
      </TestProvider>
    );

    axios.patch.mockResolvedValue({ data: { usersLiked: [] } });
    fireEvent.click(getByRole('button', { name: 'Dislike' }));
    await waitForElementToBeRemoved(getByRole('button', { name: 'Dislike' }));
  });
});
