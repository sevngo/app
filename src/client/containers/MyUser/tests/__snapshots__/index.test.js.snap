// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyUser should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-7"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 133ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiCardActionArea-root"
          href="/user/undefined"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiCardMedia-root makeStyles-media-6"
            style="background-image: url(emptyImage.png);"
          />
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5"
              />
              <h5
                class="MuiTypography-root makeStyles-age-7 MuiTypography-h5"
              >
                , 
                NaN
              </h5>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <div
          class="MuiCardActions-root"
        >
          <input
            accept="image/png, image/jpeg"
            class="makeStyles-hide-4"
            type="file"
          />
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-8"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-5"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 400ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 266ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiPaper-root makeStyles-p3-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <form
          style="width: 100%;"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                  data-shrink="true"
                >
                  New password
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                >
                  <div
                    class="MuiBox-root MuiBox-root-10 makeStyles-icon-9"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12.65 10C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H17v4h4v-4h2v-4H12.65zM7 14c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    autocomplete="password"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    name="password"
                    type="password"
                    value=""
                  />
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-11 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-13 PrivateNotchedOutline-legendNotched-14"
                    >
                      <span>
                        New password
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled MuiButton-fullWidth Mui-disabled"
                disabled=""
                tabindex="-1"
                type="submit"
              >
                <span
                  class="MuiButton-label"
                >
                  Submit
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
